alacritty/src/display/hint.rs

```diff
@@ -259,32 +371,51 @@ pub fn highlighted_at<T>(
             return None;
         }
 
-        hint.regex.with_compiled(|regex| {
-            // Setup search boundaries.
-            let mut start = term.line_search_left(point);
-            start.line = max(start.line, point.line - MAX_SEARCH_LINES);
-            let mut end = term.line_search_right(point);
-            end.line = min(end.line, point.line + MAX_SEARCH_LINES);
-
-            // Function to verify that the specified point is inside the match.
-            let at_point = |rm: &Match| *rm.end() >= point && *rm.start() <= point;
-
-            // Check if there's any match at the specified point.
-            let mut iter = RegexIter::new(start, end, Direction::Right, term, regex);
-            let regex_match = iter.find(at_point)?;
+        if hint.hyperlink {
+            if let Some(hyperlink) = term.grid()[point].hyperlink().cloned() {
+                return Some(HintMatch::Hyperlink { hyperlink, action: hint.action.clone() });
+            }
+        }
 
-            // Apply post-processing and search for sub-matches if necessary.
-            let regex_match = if hint.post_processing {
-                HintPostProcessor::new(term, regex, regex_match).find(at_point)
-            } else {
-                Some(regex_match)
-            };
+        if let Some(regex) = &hint.regex {
+            if let Some(matched) =
```

Comment:
Should never have two nested conditionals like that, it just unnecessarily increases indentation. Since the outside option isn't needed in the final scope, we can just and_then things.